hack/generator/pkg/codegen/pipeline/create_storage_types.go

```diff
@@ -61,16 +61,14 @@ func CreateStorageTypes(idFactory astmodel.IdentifierFactory) Stage {
 					continue
 				}
 
-				result.AddTypes(t)
+				storageTypes.AddTypes(t)
 			}
 
 			err := kerrors.NewAggregate(errs)
 			if err != nil {
 				return nil, err
 			}
 
-			unmodified := types.Except(result)
-			result.AddTypes(unmodified)
-			return result, nil
+			return types.OverlayWith(storageTypes), nil
```

Comment:
After re-reading the `process()` function I do indeed see that there is overlap and that we want the `storageTypes` to win because we've modified some existing `Spec` types to have an additional method.

Will leave the above here not because it's correct but because I was initially confused... wondering if a comment or a variable name change might help?

Reply:
Actually further reading the code I don't quite follow how `storageTypes` isn't just a superset of `types`? In which case there's no need for overlay you can just return `storageTypes`?

Reply:
I've collapsed the two sets into a single one just called `types`, and I've added more documentation to `process()` explaining what's going on.

Reply:
I think I'm still missing why `storageTypes` here isn't a superset of `types`?

Don't we:
1. Add all types to a `StorageTypeFactory`.
2. Return all the types we added, plus new types generated, from the `StorageTypeFactory`?

So `storageTypes already contains all the types? No `types.OverlayWith(storageTypes)` needed?

Reply:
I've gotten rid of `storage_types_factory` as it was turning into a God class; instead there are a set of individual pipeline stages that cooperate to make the required changes.